app/controllers/concerns/secure_headers_concern.rb

```diff
@@ -13,6 +18,9 @@ def apply_secure_headers_override
   def override_form_action_csp(uris)
     policy = current_content_security_policy
     policy.form_action(*uris)
+    if Identity::Hostdata.env == 'dev'
+      policy.frame_ancestors(*policy.frame_ancestors, *DEV_FRAME_ANCESTORS)
```

Comment:
I think this is the right idea in terms of overriding the CSP from a controller, but I think this is the wrong place to be doing it:

- By the method name, this is specific to the `form-action` CSP directive, and we're modifying `frame-ancestors`
- `SecureHeadersConcern` has to be included and referenced in a specific controller

To both of these points, I think it'd be good to limit the CSP modifications to the component preview routes. I think we could do something similar to how this is used, in the [`ComponentPreviewController` class](https://github.com/18F/identity-idp/blob/main/app/controllers/component_preview_controller.rb).

```rb
class ComponentPreviewController < ViewComponentsController
  if IdentityConfig.store.component_previews_enabled
    # ...
    before_action :override_frame_ancestors_csp

    def override_frame_ancestors_csp
      return if Identity::Hostdata.env != 'dev'
      policy = current_content_security_policy
      policy.frame_ancestors(*policy.frame_ancestors, *DEV_FRAME_ANCESTORS)
      request.content_security_policy = policy
    end
  end
end
```